Chrysalis Docs Review IV (#72)
iotaledger/chrysalis-docs: docs/guides/backup_security.md

# Backup and Security

> Security Checklist

Reviewer: - This text doesn't seem to describe the TOC below.
- The TOC has no links, and could possibly be deleted entirely.

Author: Agreed!

- How to backup your account.
- How to restore from a backup.
- How to export a user's Stronghold.
- How to rekey a Stronghold/password rotation.
- Do's and don'ts.

> ## Security Checklist

Reviewer: - If the checklist isn't actually checkable, I'd probably loose the checkboxes.

Author: Can be checked in the code but not in the actual page, interesting :D. Removed either way.

- [ ] I use Stronghold.
- [ ] I use a strong password (32 character length, Shannon Entropy ~ 4.0) for encrypting the stronghold.
- [ ] I rotate the stronghold password on a regular basis.
- [ ] I create a daily backup of the stronghold.snapshot file.

> - [ ] I keep a secure history of passwords used (for recovery).

Reviewer: ```suggestion
- [ ] I keep a secure history of passwords used for recovery.
 ```

- [ ] I use a secure password management service that integrates with the server.
- [ ] I use a linux based server (best memory security).

> - [ ] My server is isolated behind a DMZ.

Reviewer: ```suggestion
- [ ] I have isolated my server behind a DMZ.

> ## How to Backup Your Account

Reviewer: ```suggestion
## How to Back Up Your Account
```

Author: Surprisingly, "backup" works here.

> A simple copy of the stronghold.snapshot file works as a backup (e.g. a daily cronjob rsync/scp with a datetime suffix for example).

Reviewer: ```suggestion
You can use a copy of the `stronghold.snapshot` file as a backup.  You can implement a  [_cronjob_](https://linux.die.net/man/1/crontab), [_rsync_](https://linux.die.net/man/1/rsync) or [_scp_](https://linux.die.net/man/1/scp) with a date-time suffix to periodically back up your account.
```

Author: Updated!

## How to Restore From a Backup

> Simply place a snapshot file in your directory that wallet.rs expects.

Reviewer: ```suggestion
Simply place a snapshot file in the directory that wallet.rs expects. 
```

It'd be good to specify in which file this is configured.  Something like, "You can find the snapshot directory in the X config file under X section"

## How to Export a User's Stronghold

You can create a new Stronghold snapshot on the fly to allow a user to leave your service and retain their key.

## How to Rekey a Stronghold/Password Rotation

> The procedure for changing a Stronghold password is "simple" in that you read a snapshot into a vault and then write it out with a new encryption password. [See this code for the source.](https://github.com/iotaledger/wallet.rs/blob/d1b8893d73aae35dfcf7c5c8006e2177988d25d0/src/stronghold.rs#L436-L451)

Reviewer: ```suggestion
The procedure for changing a Stronghold password is simple.  You read a snapshot into a vault, and then write it out with a new encryption password. [See this code for the source.](https://github.com/iotaledger/wallet.rs/blob/d1b8893d73aae35dfcf7c5c8006e2177988d25d0/src/stronghold.rs#L436-L451)
```

:::Note
For obvious reasons, old snapshot backups will not be "rekeyed", so you have to track your old passwords. 

:::

> ## Do's and Don'ts

Reviewer: Seeing that there are not Do's , perhaps we could add this to the security checklist

Author: Agreed!
